cmd/create.go

```diff
@@ -0,0 +1,254 @@
+package cmd
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/orange-cloudavenue/cloudavenue-sdk-go/v1/infrapi"
+	"github.com/spf13/cobra"
+)
+
+var (
+	excrea1 = "#List all T0\ncav create vdc --name myvdc\n\n"
+	// excrea2 = "#List all T0 in wide format\ncav create t0 -o wide\n\n"
+)
+
+// getCmd represents the t0 command
+var createCmd = &cobra.Command{
+	Use:     "create",
+	Aliases: []string{"new", "add"},
+	Example: excrea1, // + excrea2,
+	Short:   "Create resource to CloudAvenue.",
+}
+
+func init() {
+	rootCmd.AddCommand(createCmd)
+
+	// ? Create command
+	createCmd.Args = cobra.NoArgs
+	// ? Create for public ip
+	createCmd.AddCommand(createPublicIPCmd)
+	// ? Create for s3
+	createCmd.AddCommand(createS3Cmd)
+	// ? Create for edgegateway
+	createCmd.AddCommand(createEdgeGatewayCmd)
+	// ? Create for vdc
+	createCmd.AddCommand(createVDCCmd)
+
+	// ? Options for edgegateway
+	createEdgeGatewayCmd.Flags().String("t0", "", "t0 name")
+	createEdgeGatewayCmd.Flags().String("vdc", "", "vdc name")
+	if err := createEdgeGatewayCmd.MarkFlagRequired("vdc"); err != nil {
+		fmt.Println("Error from Flag VDC, is require.", err)
+		return
+	}
+
+	// ? Options for publicip
+	createPublicIPCmd.Flags().String("name", "", "public ip address")
+	if err := createPublicIPCmd.MarkFlagRequired("name"); err != nil {
+		fmt.Println("Error from Flag name, is require.", err)
+		return
+	}
+
+	// ? Options for s3
+	createS3Cmd.Flags().String("name", "", "s3 bucket name")
+	if err := createS3Cmd.MarkFlagRequired("name"); err != nil {
+		fmt.Println("Error from Flag name, is require.", err)
+		return
+	}
+
+	// ? Options for vdc
+	createVDCCmd.Flags().String("name", "", "vdc name")
+	if err := createVDCCmd.MarkFlagRequired("name"); err != nil {
+		fmt.Println("Error from Flag Name, is require.", err)
+		return
+	}
+}
+
+// createCmd represents the create command
+var createPublicIPCmd = &cobra.Command{
+	Use:     "publicip",
+	Short:   "Create an ip",
+	Example: "ip create --name <>",
+
+	Run: func(cmd *cobra.Command, args []string) {
+		// Check if time flag is set
+		if cmd.Flag("time").Value.String() == "true" {
+			defer timeTrack(time.Now(), cmd.CommandPath())
+		}
```

Comment:
Cobra provide method to get bool flag

```suggestion
cmd.Flags().GetBool("time")
```

Reply:
This function returns two values to me, I find it simpler to use this little piece of code